Makefile

```
@@ -0,0 +1,20 @@
+.PHONY: all generate cargo diff
+
+all: diff
+
+generate:
+	@echo "Generating files in tools.go"
+	@cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go generate %
+
+cargo: generate
+	@cd offchain; cargo run --bin generate-schema
+	@mv offchain/schema.graphql api/graphql/reader.graphql
+
+diff: cargo
```

```suggestion
cargo: 
	@cd offchain; cargo run --bin generate-schema
	@mv offchain/schema.graphql api/graphql/reader.graphql

diff: cargo generate
```

It makes more sense to put generate as a dependency of diff instead of cargo.